src/Template/PhpTemplate.php

```diff
@@ -0,0 +1,232 @@
+            throw new RendererException($this->__('Could not retrieve template PHP vars'), 0, $e, $this);
+        }
+
+        ob_start();
+        try {
+            $this->evaluator->evaluate($vars);
+        } catch (Exception $e) {
+            // End this level of output buffering to maintain correct sequence
+            ob_end_clean();
+            throw new TemplateRenderException($this->__('Could not render template'), 0, $e, $this, $context);
+        }
+        $output = ob_get_clean();
+
+        return $output;
+    }
+
+    /**
+     * @param array|ArrayAccess|ContainerInterface $context The context to get the PHP variables for.
+     * @return array A map of PHP variable names to their values.
+     *
+     * @throws Exception If problem retrieving.
+     */
+    protected function getPhpVars($context): array
+    {
+        return [
+            static::VAR_CONTEXT => $this->getContextFunction($context),
+            static::VAR_FUNCTION => $this->getFunctionsFunction($context),
+        ];
+    }
+
+    /**
+     * Retrieves a function which itself when invoked retrieves a value from a context.
+     *
+     * That function MUST take a stringable key, and an optional default.
+     * It MUST return the value if the key is found.
+     * It MUST return the default value if the key is not found.
+     * It MAY throw {@see Exception} if there is a problem retrieving the value.
+     *
+     * @param array|ArrayAccess|ContainerInterface $context The context to get the retrieval function for.
+     *
+     * @throws Exception If problem retrieving.
+     *
+     * @return callable The function that retrieves a context value.
+     * function (string $key, mixed $default): mixed
+     */
+    protected function getContextFunction($context): callable
+    {
+        return function (string $key, $default = null) use ($context) {
+            return $this->getContextValue($context, $key, $default);
+        };
+    }
+
+    /**
+     * Retrieves a function which itself invokes a template function by name.
+     *
+     * That function MUST take a string key, and a variable list of arguments.
+     * It MUST invoke an underlying function mapped by name.
+     * It MUST pass these arguments to the underlying function , as a variable list.
+     * It MUST return the result of the invocation of the underlying function.
+     * It MAY throw an {@see Exception} if there is a problem invoking the underlying function.
+     *
+     * @param array|ArrayAccess|ContainerInterface $context The context to get the functions function for.
+     *
+     * @throws Exception If problem retrieving.
+     *
+     * @return callable The function that invokes a template function.
+     * function (string $funcName, ...$args): mixed
+     */
+    protected function getFunctionsFunction($context): callable
+    {
+        return function (string $funcName, ...$args) use ($context) {
+            return $this->invokeFunction($funcName, $args);
+        };
```

Comment:
`$context` is not used.

Reply:
Yep, it's true. However, I thought it was a good idea to pass it to all methods that create template variables, because they may want to depend on context.

What would you suggest in this case? Just remove it from the `use`?

Reply:
I'd remove the `use`. If a derived class wants to use it, the parameter is there. But in this case I wouldn't add any code that does nothing, purely for clarity and ease of mind.

Reply:
Removed `use`.